numpy/core/_dtype.py

```diff
@@ -0,0 +1,293 @@
+"""
+A place for code to be called from the implementation of np.dtype
+
+String handling is much easier to do correctly in python.
+"""
+from __future__ import division, absolute_import, print_function
+
+import numpy as np
+
+
+def __str__(dtype):
+    if dtype.fields is not None:
+        return _struct_str(dtype, include_align=True)
+    elif dtype.subdtype:
+        return _subarray_str(dtype)
```

Comment:
This line not hit in tests, see [coverage](https://codecov.io/gh/numpy/numpy/commit/73e328c9c9c4729c28f0f2dec407aed168b5fbdf#D3-15)

Reply:
Test added